server/src/main/java/com/saralein/server/request/Request.java

```diff
@@ -4,28 +4,70 @@
 
 public class Request {
     private final HashMap<String, String> request;
+    private String method;
+    private String uri;
+    private String body;
 
-    public Request(HashMap<String, String> request) {
-        this.request = request;
+    public Request(String method, String uri, String body, HashMap<String, String> headers) {
+        this.method = method;
+        this.uri = uri;
+        this.body = body;
+        this.request = headers;
     }
 
     public String getMethod() {
-        return request.get("method");
+        return method;
     }
 
     public String getUri() {
-        return request.get("uri");
+        return uri;
     }
 
     public String getBody() {
-        return request.getOrDefault("body", "");
+        return body;
     }
 
-    public String getAuthorization() {
-        return request.getOrDefault("Authorization", "");
+    public String getHeader(String key) {
+        return request.getOrDefault(key, "");
     }
 
     public String getRequestLine() {
         return String.format("%s %s HTTP/1.1", getMethod(), getUri());
     }
+
+    public static class Builder {
+        private String method = "";
+        private String uri = "";
+        private String body = "";
+        private HashMap<String, String> headers = new HashMap<>();
+
+        public Builder addMethod(String method) {
```

Comment:
Makes sense when you explain it like that.  I've renamed these methods for the response, request, and header.  I've left the methods `addHeader` or `addHeaders` alone since those are additive.